sdk/go/common/workspace/plugins.go

```diff
@@ -706,6 +801,10 @@ func (spec PluginSpec) GetSource() (PluginSource, error) {
 				return newGithubSource(url, spec.Name, spec.Kind)
 			}
 
+			if url.Scheme == "gitlab" {
+				return newGitlabSource(url, spec.Name, spec.Kind)
+			}
```

Comment:
Optional style tweak: This might be cleaner as a switch statement:

```go
switch url.Scheme {
case "github"
  return newGithubSource(..)
case "gitlab":
  return newGitlabSource(..)
default:
  return newPluginURLSource(..)
```

In the future, we could even turn this into a plugin source registry of some kind, so that it's not a hard-coded list.